application-analytics-default/src/main/java/com/xwiki/analytics/internal/MatomoAnalyticsManager.java

```diff
@@ -0,0 +1,115 @@
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.analytics.internal;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.ws.rs.core.UriBuilder;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.manager.ComponentLookupException;
+import org.xwiki.resource.CreateResourceReferenceException;
+import org.xwiki.resource.CreateResourceTypeException;
+import org.xwiki.resource.UnsupportedResourceReferenceException;
+import org.xwiki.stability.Unstable;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.xwiki.analytics.AnalyticsManager;
+import com.xwiki.analytics.JsonNormaliser;
+import com.xwiki.analytics.configuration.AnalyticsConfiguration;
+
+/**
+ * The class would handle the Matomo request and return a json with the data.
+ *
+ * @version $Id$
+ * @since 1.0
+ */
+@Component
+@Named("Matomo")
+@Singleton
+@Unstable
+public class MatomoAnalyticsManager implements AnalyticsManager
+{
+    @Inject
+    @Named("MostViewedPages")
+    private JsonNormaliser mostViewedNormaliser;
+
+    @Inject
+    private AnalyticsConfiguration configuration;
+
+    /**
+     * This method will handle all the request made by the user and return a proper json.
+     *
+     * @param jsonNormaliserHint hint to select the json normaliser.
+     * @param parameters a list of key, value pairs that will represent the parameters for the request.
+     * @return will return a json with all the data returned by Matomo.
+     */
+    @Override
+    public JsonNode requestData(Map<String, String> parameters, String jsonNormaliserHint)
+        throws IOException, InterruptedException, ComponentLookupException, UnsupportedResourceReferenceException,
+        CreateResourceTypeException, CreateResourceReferenceException
+    {
+        parameters.put("idSite", configuration.getIdSite());
+        parameters.put("token_auth", configuration.getAuthenticationToken());
+        JsonNormaliser jsonNormaliser = this.getNormaliser(jsonNormaliserHint);
+        if (jsonNormaliser == null) {
+            throw new RuntimeException("The hint that you provided is invalid.");
```

Comment:
`throw new RuntimeException("Error while getting Matomo statistic results .");`. What happens on the UI side when this error is thrown?
Also, I am thinking now that this hint is something that we control from the velocity code, so it should never be null. I'm not sure what is best to do, but maybe you could just log a warning for the developer to know

Reply:
Hello!
This is what the user will see when the error occurs. I added some extra details to let the user know why the error happened. I think that this message will enough for a developer to understand that the hint can't be null.
![image](https://github.com/xwikisas/application-analytics/assets/92780090/e2b90f50-e559-481a-8141-64061bc79768)